gen/internals.py

```diff
@@ -275,7 +275,7 @@ def make_id(self):
 # NOTE: This exception should never escape the Resolver
 class CalculatorError(Exception):
 
-    def __init__(self, message: str, chain: Optional[list]):
```

Comment:
The change could simply be:

chain = None

Ref: https://www.python.org/dev/peps/pep-0484/#union-types
```
An optional type is also automatically assumed when the default value is None , for example:

def handle_employee(e: Employee = None): ...
This is equivalent to:

def handle_employee(e: Optional[Employee] = None) -> None: ...
```